src/test/java/tr/edu/ogu/ceng/service/StudentTest.java

```diff
@@ -34,4 +40,17 @@ public void when_resultIsNotNull_then_returnValidDTO() {
 
 	}
 
+	@Test
+	void should_save_one_student() {
+		final var studentToSave = Student.builder().name("Name").surname("Surname").id(6L).build();
+
+		when(studentRepository.save(any(Student.class))).thenReturn(studentToSave);
+
+		final var actual = studentService.addStudent(new Student());
```

Comment:
Bu testin fail olmasını bekleriz, neden? Çünkü zorunlu alanlar var ve bu notNull alanlar mutlaka parametre veriliyor olmalı. Bunlar olmadan verilen bir test fail etmeli. any(Student.class) bu kapsamda uygun görünmedi bana.

Reply:
Bununla birlikte bu şu demek oluyor. Repository dönüyor olabilir ama servis katmanında ilgili kontrollerin yapılması bekleniyor. Ya da controllera daha gelmeden önce ilgili değerleri aldığının doğrulandığına dair testlerin eklenmesi gerekir. Sonuç olarak aslında bu test senaryosu başarılı olarak geçmek için çok uygun görünmüyor. Ne olursan ol yeşil ol denmiş gibi duruyor.